File: problem_solving_for_beginners/bitwise/c-programming/check_nth_bit.c
#include<stdio.h>
/***
 * main -Checks nth bit if it's set or unset using bitwise
 * 
 * description: Bitwise LEFT SHIFT is used to shift to nth bit to check
 * then bitwise AND to check. zero (0) is return if it's unset 
 * otherwise any other number aside from zero
 * 
 * Return: prints ON otherwise OFF
 */
 
 int main(void)
 {  
     int num, n, res;
     
     printf("Enter number and nth bit you want to enable: ");
     scanf("%d%d", &num, &n);


    	
	res = num & (1 << (n - 1));
	
	if (res)
	    printf("ON\n");
	else
	    printf("OFF\n");
	    
	return 0;
}

File: problem_solving_for_beginners/bitwise/python/check_nth_bit.py
#!/usr/bin/python3
"""
    Checks nth bit if it's set or unset using 
    bitwise OR and LEFT SHIFT operators
"""

num, n = input("Enter Number and nth bit to enable: ").split()
num = int(num)
n = int(n) - 1

res = num & (1 << n)

if res:
    print("ON")
else:
    print("OFF")